Sources/Web3Core/KeystoreManager/BIP32Keystore.swift

```diff
@@ -176,29 +176,29 @@ public class BIP32Keystore: AbstractKeystore {
             if let upperIndex = (path.range(of: prefixPath)?.upperBound), upperIndex < path.endIndex {
                 pathAppendix = String(path[path.index(after: upperIndex)..<path.endIndex])
             } else {
-                throw AbstractKeystoreError.encryptionError("out of bounds")
+                throw AbstractKeystoreError.encryptionError("BIP32Keystore. Given derivation path has no extra sections after the rootPrefix. Root prefix: '\(rootPrefix)'; given derivation path: '\(path)'. Expected to be true: `path.range(of: rootPrefix).upperBound < path.endIndex`.")
             }
         }
         if pathAppendix.hasPrefix("/") {
             pathAppendix = pathAppendix.trimmingCharacters(in: .init(charactersIn: "/"))
         }
         guard rootNode.depth == prefixPath.components(separatedBy: "/").count - 1 else {
-            throw AbstractKeystoreError.encryptionError("Derivation depth mismatch")
+            throw AbstractKeystoreError.encryptionError("BIP32Keystore. Derivation depth mismatch.")
         }
         guard let newNode = rootNode.derive(path: pathAppendix, derivePrivateKey: true) else {
-            throw AbstractKeystoreError.keyDerivationError
+            throw AbstractKeystoreError.keyDerivationError("BIP32Keystore. Failed to derive a new node. Check given parent node and path.")
```

Comment:
We really need to revisit all our errors and make sure:
1) We have description;
2) The description makes sense to the developers who could get this error.